sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/json/JsonFiltersSuite.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.json
+
+import org.apache.spark.sql.catalyst.{StructFilters, StructFiltersSuite}
+import org.apache.spark.sql.sources
+import org.apache.spark.sql.types.StructType
+
+class JsonFiltersSuite extends StructFiltersSuite {
+  def createFilters(filters: Seq[sources.Filter], schema: StructType): StructFilters = {
+    new JsonFilters(filters, schema)
+  }
```

Comment:
The tests will not check anything affected by this PR because filters w/ nested attrs are not pushed down to JSON datasource.

Reply:
Ah, I see. I noticed now that nested filtered are filtered out in advance.

Reply:
Do you think it's possible to implement nested situations? Although JSON is not columnar format, but if we are able to push down nested predicate, we will be able to avoid de-serialize unnecessary data.

Reply:
> Do you think it's possible to extend to nested cases?

I think it is possible. I see just a few difficulties that should be solved:
1. `JacksonParser`.`convertObject` sees the current row only. Need to pass all parent rows, set current one at right position at the parent row to apply filters.
2. Currently, JacksonParser doesn't allow to stop parsing inside of **nested** objects and skip the rest. I mean normally skip without throwing exceptions like `PartialResultException`. 
3. Need to more effectively restrict the number of filters that are applied while digging into nested objects. 
4. Rewrite JsonFilters and replace arrays by hash maps, probably or maybe not. That's need to try. Current implementation is based indexes bound to schema.
5. This method `def skipRow(row: InternalRow, index: Int): Boolean` should be changed. `row` should point out to the top row. And `index` should be replaced to "full" path to attr.

Reply:
Get you. Seems it's non-trivial. Let's iterate it once this is merged. Thank you very much!